src/hooks/useRewards.ts

```diff
@@ -0,0 +1,60 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { RewardOption } from '@windingtree/glider-types/types/win';
+import axios from 'axios';
+import { backend } from 'src/config';
+
+type MutationProps = {
+  id?: string | null;
+  rewardType?: string;
+};
+
+interface ClaimRewardResponse {
+  success: boolean;
+}
+
+const getRewards = async (id: string) => {
+  const { data } = await axios
+    .get<RewardOption[]>(`${backend.url}/api/booking/${id}/rewardOptions`)
+    .catch((_) => {
+      throw new Error('Could not retrieve rewards');
+    });
+
+  return data;
+};
+
+const postClaimReward = async ({ id, rewardType }: MutationProps) => {
+  const { data } = await axios
+    .post<ClaimRewardResponse>(`${backend.url}/api/booking/${id}/reward`, {
+      rewardType
+    })
+    .catch((_) => {
+      throw new Error('Something went wrong with claiming your reward.');
+    });
+
+  if (!data.success) {
+    throw new Error('Something went wrong with claiming your reward.');
+  }
+
+  return data;
+};
+
+export const useRewards = (id: string | null) => {
```

Comment:
please add types, the values returned from hook should be typed. As reference  `UseBooking`, `UseBookingAuth`